File: Python/696.py
class Solution(object):
    def countBinarySubstrings(self, s):
        n = len(s)
        lst = []
        k = 0
        count = 0
        for i in range(1,n):
            if s[i] != s[i-1]:
                lst.append(s[k:i])
                k = i
        lst.append(s[k:])
        for i in range(1,len(lst)):
            if lst[i-1][0] != lst[i][0]:
                m = len(lst[i-1])
                n = len(lst[i])
                count += min(m,n)
        return count

File: Java/696.java
import java.util.*;
class Solution {
    public int countBinarySubstrings(String s) {
        ArrayList<String> lst = new ArrayList<>();
        int k = 0, count = 0, a = 0, b = 0;
        int n = s.length();
        for(int i = 1; i < n; i++){
            if(s.charAt(i)!=s.charAt(i-1)){
                lst.add(s.substring(k,i));
                k = i;
            }
        }
        lst.add(s.substring(k));
        for(int i = 1; i < lst.size(); i++){
            if(lst.get(i-1).charAt(0) != lst.get(i).charAt(0)){
                a = lst.get(i-1).length();
                b = lst.get(i).length();
                if(a>b) count += b;
                else count += a;
            }
        }
        return count;
    }
}